python/lsst/daf/butler/gen2convert/walker.py

```diff
@@ -199,6 +240,16 @@ def scanRepo(self, repo):
                     log.debug("%s: found %s with %s", repo.root, dataset.datasetType.name, dataset.dataId)
                     repo.datasets[dataset.datasetType.name][filePath] = dataset
 
+                    try:
+                        calibRow = self._calibDict[(
+                            dataset.datasetType.name, dataset.dataId["calibDate"], dataset.dataId["ccd"])]
+                        if calibRow["filter"] is None or calibRow["filter"] == dataset.dataId["filter"]:
+                            dataset.dataId["valid_first"] = calibRow["valid_first"]
+                            dataset.dataId["valid_last"] = calibRow["valid_last"]
+                            print(calibRow)
+                    except KeyError:
```

Comment:
Instead of using `KeyError` to do control flow for the common case where the dataset is not a calibration product, I'd recommend using something like:
```
calibRow = self._calibDict.get(..., None)
if calibRow is not None:
    ...
```
That's just a slight preference, and possibly one driven by experience with languages that approach exceptions differently from Python.  Feel free to push back if you disagree.

Reply:
I'm fine with this.  Using the exception was something I wasn't totally happy with, so a default value works nicely.